server/src/main/java/org/elasticsearch/index/shard/IndexShard.java

```diff
@@ -1355,8 +1356,11 @@ public void flush(FlushRequest request) {
          */
         verifyNotClosed();
         final long time = System.nanoTime();
-        getEngine().flush(force, waitIfOngoing);
-        flushMetric.inc(System.nanoTime() - time);
+        boolean flushHappened = getEngine().flush(force, waitIfOngoing);
+        if (flushHappened) {
```

Comment:
I am in doubt about this. The [documentation](https://www.elastic.co/guide/en/elasticsearch/reference/current/cluster-nodes-stats.html#cluster-nodes-stats-api-response-body) says this is "Number of flush operations". So far it has been the same as API calls, now it is only those that do anything. It is a subtle change and hardly breaking.

But this change puts the flush-metric between API calls and actual flushes. It is not actual flushes, because if there is nothing to flush, we will still increase it here.

So I'd prefer to inc always here and then we can always make it more accurate.

Reply:
Done